src/test/engine-integrations/cross-margin/PreviewCollateralReq.t.sol

```diff
@@ -16,219 +16,446 @@ import "../../../core/engines/cross-margin//AccountUtil.sol";
+        OptionPosition[] memory positions = new OptionPosition[](2);
+        positions[0] = OptionPosition(TokenType.PUT, 17999_999999, 1 * sUNIT);
+        positions[1] = OptionPosition(TokenType.PUT, 18000 * UNIT, -1 * sUNIT);
 
         Balance[] memory balances = _previewMinCollateral(positions);
 
         assertEq(balances.length, 1);
         assertEq(balances[0].collateralId, usdcId);
-        assertEq(balances[0].amount, 28000 * 1e6);
+        assertEq(balances[0].amount, 1);
     }
 
-    // function testPreviewMinCollateralCallsPut4() public {
-    //     Position[] memory shorts;
-    //     Position[] memory longs;
+    function testMarginCallSpreadSameUnderlyingCollateral() public {
+        OptionPosition[] memory positions = new OptionPosition[](2);
+        positions[0] = OptionPosition(TokenType.CALL, 21999 * UNIT, -1 * sUNIT);
+        positions[1] = OptionPosition(TokenType.CALL, 22000 * UNIT, 1 * sUNIT);
 
-    //     oracle.setSpotPrice(address(weth), 19000 * UNIT);
+        Balance[] memory balances = _previewMinCollateral(positions);
 
-    //     uint256 p17000 = getTokenId(TokenType.PUT, pidUsdcCollat, expiry, 17000 * UNIT, 0);
-    //     uint256 p18000 = getTokenId(TokenType.PUT, pidUsdcCollat, expiry, 18000 * UNIT, 0);
+        assertEq(balances.length, 1);
+        assertEq(balances[0].collateralId, wethId);
+        assertEq(balances[0].amount, ((1 * UNIT) / 22000) * 1e12);
+    }
 
-    //     uint256 c21000 = getTokenId(TokenType.CALL, pidEthCollat, expiry, 21000 * UNIT, 0);
-    //     uint256 c22000 = getTokenId(TokenType.CALL, pidEthCollat, expiry, 22000 * UNIT, 0);
-    //     uint256 c25000 = getTokenId(TokenType.CALL, pidEthCollat, expiry, 25000 * UNIT, 0);
-    //     uint256 c26000 = getTokenId(TokenType.CALL, pidEthCollat, expiry, 26000 * UNIT, 0);
-    //     uint256 c27000 = getTokenId(TokenType.CALL, pidEthCollat, expiry, 27000 * UNIT, 0);
+    function testMarginCallSpreadSameUnderlyingCollateralBiggerNumbers() public {
+        OptionPosition[] memory positions = new OptionPosition[](2);
+        positions[0] = OptionPosition(TokenType.CALL, 21000 * UNIT, -100000 * sUNIT);
+        positions[1] = OptionPosition(TokenType.CALL, 22000 * UNIT, 100000 * sUNIT);
 
-    //     shorts = new Position[](5);
-    //     shorts[0] = Position(p18000, uint64(1 * UNIT));
-    //     shorts[1] = Position(c21000, uint64(1 * UNIT));
-    //     shorts[2] = Position(c22000, uint64(8 * UNIT));
-    //     shorts[3] = Position(c26000, uint64(8 * UNIT));
-    //     shorts[4] = Position(c27000, uint64(1 * UNIT));
+        Balance[] memory balances = _previewMinCollateral(positions);
 
-    //     longs = new Position[](2);
-    //     longs[0] = Position(p17000, uint64(1 * UNIT));
-    //     longs[1] = Position(c25000, uint64(16 * UNIT));
+        assertEq(balances.length, 1);
+        assertEq(balances[0].collateralId, wethId);
+        assertEq(balances[0].amount, ((100000 * 1000 * UNIT) / 22000) * 1e12);
```

Comment:
@Fletch153 what currency is `1e12`?

Reply:
I used this formula to calculate how much ETH collateral is required (4 545 454 545 000 000 000 000), as the original formula in CMM.t.sol ((1000_00000 * **10^6**) / 22000) was based on the underlying having a precision of 10^6, but altering it to (1000_00000 * **10^18** / 22000) gives a result of '4 545 454 545 454 545 454 545', which won't work due to precision limits within Grappa.